src/main/java/com/sourcegraph/webhook/WebhookRouter.java

```diff
@@ -0,0 +1,63 @@
+package com.sourcegraph.webhook;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.stream.JsonReader;
+import com.sourcegraph.webhook.registry.Webhook;
+import com.sourcegraph.webhook.registry.WebhookException;
+import com.sourcegraph.webhook.registry.WebhookRegistry;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.*;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+
+@Path("/webhook")
+public class WebhookRouter {
+    private static final Logger log = LoggerFactory.getLogger(WebhookRouter.class);
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response get() {
+        List<Webhook> hooks = WebhookRegistry.getWebhooks();
+        String resp = new Gson().toJson(hooks);
+        return Response.ok(resp).build();
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response put(String raw) throws WebhookException {
```

Comment:
We'll need some basic documentation on how to use this API (but that can come in a follow up PR).

Reply:
Question that comes to mind: Are requests authenticated in the same way as the normal Bitbucket Server API?

Reply:
Yep!

```
Authentication
Any authentication that works against Bitbucket will work against the REST API. The preferred authentication methods are HTTP Basic (when using SSL) and OAuth. Other supported methods include: HTTP Cookies and Trusted Applications.

You can find OAuth code samples in several programming languages at bitbucket.org/atlassian_tutorial/atlassian-oauth-examples.

The log-in page uses cookie-based authentication, so if you are using Bitbucket in a browser you can call REST from JavaScript on the page and rely on the authentication that the browser has established.
```
(https://docs.atlassian.com/bitbucket-server/rest/5.16.0/bitbucket-rest.html)

Reply:
This talks about the REST API. Does this plugin "extend" the same REST API? Or is it served under a different path? I guess all of these usage questions need to be documented.

Reply:
Yes, I believe the plugin extend bitbucket REST API. They're both served under the `/rest/` path which requires authentication (else 401s).